logstash-core/lib/logstash/agent.rb

```diff
@@ -33,7 +33,7 @@ def execute
   end # def execute
 
   def add_pipeline(pipeline_id, config_str, settings = {})
-    @pipelines[pipeline_id] = LogStash::Pipeline.new(config_str, settings)
+    @pipelines[pipeline_id] = LogStash::Pipeline.new(config_str, settings.merge(:pipeline_id => pipeline_id))
```

Comment:
@andrewvc this pipeline_id handling is quirky,I see that in pipeline.rb if the id is nil then `object_id` is used. maybe we could do it here instead? because otherwise calling `add_pipeline(nil, "") will create a nil key in @pipelines but the pipeline.id will return `object_id`